packages/flutter/test/material/outline_button_test.dart

```diff
@@ -0,0 +1,252 @@
+        ..clipPath(pathMatcher: coversSameAreaAs(clipPath, areaToCompare: clipRect.inflate(10.0)))
+        ..path(color: borderColor, strokeWidth: borderWidth)
+    );
+  });
+
+
+  testWidgets('OutlineButton contributes semantics', (WidgetTester tester) async {
+    final SemanticsTester semantics = new SemanticsTester(tester);
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new Material(
+          child: new Center(
+            child: new OutlineButton(
+              onPressed: () { },
+              child: const Text('ABC')
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(semantics, hasSemantics(
+      new TestSemantics.root(
+        children: <TestSemantics>[
+          new TestSemantics.rootChild(
+            actions: <SemanticsAction>[
+              SemanticsAction.tap,
+            ],
+            label: 'ABC',
+            rect: new Rect.fromLTRB(0.0, 0.0, 88.0, 36.0),
+            transform: new Matrix4.translationValues(356.0, 282.0, 0.0),
+            flags: <SemanticsFlag>[
+              SemanticsFlag.isButton,
+              SemanticsFlag.hasEnabledState,
+              SemanticsFlag.isEnabled,
+            ],
+          )
+        ],
+      ),
+      ignoreId: true,
+    ));
+
+    semantics.dispose();
+  });
+
+
+  testWidgets('OutlineButton scales textScaleFactor', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new Material(
+          child: new MediaQuery(
+            data: const MediaQueryData(textScaleFactor: 1.0),
+            child: new Center(
+              child: new OutlineButton(
+                onPressed: () { },
+                child: const Text('ABC'),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byType(OutlineButton)), equals(const Size(88.0, 36.0)));
+    expect(tester.getSize(find.byType(Text)), equals(const Size(42.0, 14.0)));
+
+    // textScaleFactor expands text, but not button.
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new Material(
+          child: new MediaQuery(
+            data: const MediaQueryData(textScaleFactor: 1.3),
+            child: new Center(
+              child: new FlatButton(
+                onPressed: () { },
+                child: const Text('ABC'),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byType(FlatButton)), equals(const Size(88.0, 36.0)));
+    // Scaled text rendering is different on Linux and Mac by one pixel.
+    // TODO(#12357): Update this test when text rendering is fixed.
+    expect(tester.getSize(find.byType(Text)).width, isIn(<double>[54.0, 55.0]));
+    expect(tester.getSize(find.byType(Text)).height, isIn(<double>[18.0, 19.0]));
```

Comment:
FYI: see also [within](https://docs.flutter.io/flutter/flutter_test/within.html) utility, which supports `Size`.

Reply:
That might make the test clearer however I'm going to leave this alone for now because it matches similar code in buttons_test.dart that will get update when #12357 is fixed.